packages/cdktf-cli/lib/execution-logs.ts

```diff
@@ -0,0 +1,68 @@
+import { logger } from "./logging";
+import { SynthesizedStack } from "./synth-stack";
+type TFIdentifier = string;
+type CDKTFResourcePath = string;
+type TfResource = Record<
+  string,
+  {
+    ["//"]?: {
+      metadata?: {
+        path?: string;
+      };
+    };
+  }
+>;
+type TfResourceType = Record<string, TfResource>;
+type TFJson = {
+  data: TfResourceType;
+  resource: TfResourceType;
+};
+
+export function enhanceLogMessage(
+  message: string,
+  stack: SynthesizedStack
+): string | undefined {
+  // we never want to throw, if it does not work we just do as if it did not happen
+  try {
+    const json = JSON.parse(stack.content) as TFJson;
+    const pathMapping: Record<TFIdentifier, CDKTFResourcePath> = {};
+
+    ["data", "resource"].forEach((type) => {
+      Object.entries(json[type as "data" | "resource"] || {}).forEach(
+        ([resourceType, resourceInstances]) => {
+          Object.entries(resourceInstances || {}).forEach(
+            ([resourceName, resource]) => {
+              // Some constructs don't have this metadata
+              if (
+                resource["//"] &&
+                resource["//"].metadata &&
+                resource["//"].metadata.path
+              ) {
+                pathMapping[`${resourceType}.${resourceName}`] = resource[
+                  "//"
+                ].metadata.path.replace(`${stack.name}/`, "");
+              }
+            }
+          );
+        }
+      );
+    });
```

Comment:
Could we cache this? Maybe by transforming `enhanceLogMessage(message, stack) => string | undefined` to a higher-order function like `createEnhanceLogMessage(stack) => (message) => string | undefined`. Then we wouldn't need to parse the json on every message but rather only once to create the pathMapping?

Reply:
Yeah 👍